src/msg/async/Stack.cc

```diff
@@ -41,7 +41,7 @@ std::function<void ()> NetworkStack::add_thread(unsigned i)
       char tp_name[16];
       sprintf(tp_name, "msgr-worker-%u", w->id);
       ceph_pthread_setname(pthread_self(), tp_name);
-      const uint64_t EventMaxWaitUs = 30000000;
+      const int EventMaxWaitUs = 30000000;
```

Comment:
instead, i think we should change `EventCenter::process_events(int timeout_microseconds,  ceph::timespan *working_dur)` to `EventCenter::process_events(unsigned timeout_microseconds,  ceph::timespan *working_dur)` . as the only reason we need an `int` timeout_microseconds is that `std::chrono::microseconds::rep` is a signed integer, but here, what we want is but a timeout, which will never be less than zero.

Reply:
that's ok, it would look better.